frontend/model/contracts/group.js

```diff
@@ -94,6 +94,7 @@ function updateCurrentDistribution ({ contractID, meta, state, getters }) {
   const noPayments = Object.keys(curPeriodPayments.paymentsFrom).length === 0
   // update distributionDate if we've passed into the next period
   if (comparePeriodStamps(period, getters.groupSettings.distributionDate) > 0) {
+    updateGroupStreaks({ state, getters })
```

Comment:
You probably want to call `updateGroupStreaks` after `updateAdjustedDistribution` is called below, since that will save and update `lastAdjustedDistribution`

Reply:
@taoeffect the reason why it is placed here is, I thought it would make sense to check & update the streaks status re payments right before the group's distribution period is updated to the next one, which is right below this line.
But like you mentioned in the above review comment, I can add logics where it uses the same trick in `groupIncomeAdjustedDistribution ` for a case when `lastAdjustedDistribution` has no value. :)

Reply:
@taoeffect question here, I was actually looking for a place where the group's distribution period is set to the next one and all the state related to the payments are reset accordingly. does it seems like here(inside `updateCurrentDistribution()` and within `comparePeriodStamps(period, getters.groupSettings.distributionDate) > 0` block) is a correct place in that regard?

Reply:
@SebinSong There are two places you might consider: one is in the Vuex getters (see `groupIncomeAdjustedDistribution` and how it relies on both `thisPeriodPaymentInfo` and `lastAdjustedDistribution`), and the other is either in `updateCurrentDistribution` or `updateAdjustedDistribution`.

I would look at all these places carefully to see how they're called and make a decision.

Reply:
>  thought it would make sense to check & update the streaks status re payments right before the group's distribution period is updated to the next one, which is right below this line.

OK, I don't understand the reasoning for this though.

Doing it this way results in a DRY violation because you end up running the `adjustedDistribution` algorithm twice. So could we instead move this code below so that we avoid the DRY violation? Or is there some downside I'm not seeing?